src/PhpDoc/TypeNodeResolver.php

```diff
@@ -311,6 +313,14 @@ private function resolveGenericTypeNode(GenericTypeNode $typeNode, NameScope $na
 		}
 
 		$mainType = $this->resolveIdentifierTypeNode($typeNode->type, $nameScope);
+		$broker = Broker::getInstance();
+
+		if ($mainType instanceof TypeWithClassName && $broker->hasClass($mainType->getClassName())) {
+			$classReflection = $broker->getClass($mainType->getClassName());
+			if ($classReflection->isGeneric()) {
+				return new GenericObjectType($mainType->getClassName(), $genericTypes);
+			}
+		}
```

Comment:
Here, I want to preserve the existing behaviour around iterables:

- If the type is known to be generic, create a `GenericObjectType`
- If it's known to be iterable, create a `ObjectType&iterable<>` (existing behaviour)
- Fallback to creating a `GenericObjectType` if it's a `TypeWithClassName`

Reply:
Yes, agreed :)